spec/MongoStorageAdapter.spec.js

```diff
@@ -286,4 +286,29 @@ describe_only_db('mongo')('MongoStorageAdapter', () => {
       done();
     });
   });
+
+  it('getClass if exists', async () => {
+    const adapter = new MongoStorageAdapter({ uri: databaseURI });
+
+    const schema = {
+      fields: {
+        array: { type: 'Array' },
+        object: { type: 'Object' },
+        date: { type: 'Date' },
+      },
+    };
+
+    await adapter.createClass('MyClass', schema);
+    const myClassSchema = await adapter.getClass('MyClass');
+    expect(myClassSchema).toBeDefined();
+  });
+
+  it('getClass if not exists', async () => {
```

Comment:
I'm still confused by this.

is it even possible for a catch to have an undefined argument?

is it supposed to go into the catch block?

confused.

Reply:
```
_fetchOneSchemaFrom_SCHEMA(name: string) {
    return this._collection
      ._rawFind(_mongoSchemaQueryFromNameQuery(name), { limit: 1 })
      .then(results => {
        if (results.length === 1) {
          return mongoSchemaToParseSchema(results[0]);
        } else {
          throw undefined;
        }
      });
  }
```

catch just handles whatever is thrown.

Reply:
It is weird, but it really needs to throw undefined in order to this line to work: https://github.com/parse-community/parse-server/blob/master/src/Controllers/SchemaController.js#L1040

Reply:
This test covers this case in the API level (https://github.com/parse-community/parse-server/blob/8f0e7f3bcc567f1ae5218df9b22b0b34d3db4930/spec/schemas.spec.js#L269). So if it is passing, we should be fine.

Reply:
Anyway I'd improve the test to something like this:
```
await expectAsync(adapter.getClass('UnknownClass')).toBeRejectedWith(undefined);
```

Reply:
agreed that if it is passing, its ok to get this pr in.

and the `throw undefined` now makes it make sense to me.  I've just never come across such a construct.  its strange.